src/vos/vos_dtx.c

```diff
@@ -1557,7 +1557,10 @@ vos_dtx_deregister_record(struct umem_instance *umm, daos_handle_t coh,
 	if (rec_df == NULL)
 		return;
 
-	for (i = 0; i < dae_df->dae_rec_cnt - DTX_INLINE_REC_CNT; i++) {
+	count = dae_df->dae_rec_cnt > DTX_INLINE_REC_CNT;
```

Comment:
This code is confusing.  On this line count is set to result of a boolean expression.   Is there some compiler/C magic we are counting on here?   If so, I think it should be rewritten to be easier to read.

Reply:
Sorry, that is typo.